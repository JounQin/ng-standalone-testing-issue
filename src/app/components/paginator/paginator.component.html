<aui-select>
  <aui-option *ngFor="let size of pageSizeOptions" [value]="size">
    {{ size }}
  </aui-option>
</aui-select>
