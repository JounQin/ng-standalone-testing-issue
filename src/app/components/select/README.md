## Select

### Select Attributes

| 参数        | 说明                                         | 类型               | 默认值                  |
| ----------- | -------------------------------------------- | ------------------ | ----------------------- |
| size        | 尺寸                                         | ComponentSize      | ComponentSize.Medium    |
| placeholder | 占位符                                       | string             | -                       |
| disabled    | 禁用                                         | boolean            | false                   |
| loading     | 加载状态                                     | boolean            | false                   |
| value       | 选中值                                       | T                  | -                       |
| filterable  | 是否可过滤选项                               | boolean            | false                   |
| filterFn    | 过滤函数                                     | OptionFilterMethod | Array.property.includes |
| allowCreate | 是否允许创建新值，需要和 filterable 一起打开 | boolean            | false                   |
| clearable   | 是否可清空选项                               | boolean            | false                   |
| trackFn     | 返回 value 唯一标识符                        | TrackFn            | val => val              |

### Select Events

| 时间名称     | 说明         | 回调参数      |
| ------------ | ------------ | ------------- |
| valueChange  | 选中值变化   | value: T      |
| filterChange | input 值变化 | value: string |

## MultiSelect

### MultiSelect Attributes

| 参数        | 说明                                         | 类型           | 默认值                  |
| ----------- | -------------------------------------------- | -------------- | ----------------------- |
| size        | 尺寸                                         | ComponentSize  | ComponentSize.Medium    |
| placeholder | 占位符                                       | string         | -                       |
| disabled    | 禁用                                         | boolean        | false                   |
| loading     | 加载状态                                     | boolean        | false                   |
| value       | 选中值                                       | T[]            | []                      |
| filterable  | 是否可过滤选项                               | boolean        | false                   |
| filterFn    | 过滤函数                                     | OptionFilterFn | Array.property.includes |
| allowCreate | 是否允许创建新值，需要和 filterable 一起打开 | boolean        | false                   |
| clearable   | 是否可清空选项                               | boolean        | false                   |
| trackFn     | 返回 value 唯一标识符                        | TrackFn        | val => val              |

### MultiSelect Events

| 事件名称     | 说明         | 回调参数      |
| ------------ | ------------ | ------------- |
| valueChange  | 选中值变化   | value: T      |
| filterChange | input 值变化 | value: string |

## Option

### Option Attributes

| 参数     | 说明                     | 类型    | 默认值 |
| -------- | ------------------------ | ------- | ------ |
| label    | 选中后 select 显示的文案 | string  | -      |
| value    | 选中值                   | T       | -      |
| disabled | 禁用                     | boolean | false  |
