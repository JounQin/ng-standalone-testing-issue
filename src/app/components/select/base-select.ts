import { ContentChildren, Directive, QueryList } from '@angular/core';

import { OptionComponent } from './option/option.component';

@Directive({
  standalone: true,
})
export abstract class BaseSelect<T, V = T> {
  @ContentChildren(OptionComponent, { descendants: true })
  contentOptions: QueryList<OptionComponent<T>>;
}
