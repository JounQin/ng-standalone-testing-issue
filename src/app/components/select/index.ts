export * from './option/option.component';
export * from './select.component';
export * from './select.types';
export * from './validators';
