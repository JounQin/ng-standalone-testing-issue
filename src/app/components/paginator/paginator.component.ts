import { NgFor } from '@angular/common';
import {
  ChangeDetectionStrategy,
  Component,
  ViewEncapsulation,
} from '@angular/core';

import { OptionComponent } from '../select/option/option.component';
import { SelectComponent } from '../select/select.component';

@Component({
  selector: 'aui-paginator',
  templateUrl: 'paginator.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None,
  preserveWhitespaces: false,
  standalone: true,
  imports: [NgFor, SelectComponent, OptionComponent],
})
export class PaginatorComponent {
  pageSizeOptions = [20, 50, 100];
}
