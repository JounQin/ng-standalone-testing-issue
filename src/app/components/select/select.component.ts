import {
  ChangeDetectionStrategy,
  Component,
  ViewEncapsulation,
  forwardRef,
} from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';

import { BaseSelect } from './base-select';

@Component({
  selector: 'aui-select',
  templateUrl: './select.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
  encapsulation: ViewEncapsulation.None,
  preserveWhitespaces: false,
  providers: [
    {
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => SelectComponent),
      multi: true,
    },
    {
      provide: BaseSelect,
      useExisting: SelectComponent,
    },
  ],
  standalone: true,
  imports: [],
})
export class SelectComponent<T = unknown> extends BaseSelect<T> {}
