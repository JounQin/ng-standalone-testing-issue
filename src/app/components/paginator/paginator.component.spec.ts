import { Component } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { SelectComponent } from '../select/select.component';

import { PaginatorComponent } from './paginator.component';

describe('PaginatorComponent', () => {
  let fixture: ComponentFixture<TestComponent>;

  beforeEach(() => {
    fixture = TestBed.createComponent(TestComponent);
    fixture.detectChanges();
  });

  it('should render current template', () => {
    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  template: `
    <aui-paginator />
  `,
  standalone: true,
  imports: [
    PaginatorComponent,
    // SelectComponent
  ],
})
class TestComponent {}
