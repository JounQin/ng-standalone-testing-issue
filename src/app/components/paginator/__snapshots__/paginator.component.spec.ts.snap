// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatorComponent should render current template 1`] = `
<ng-component>
  <aui-paginator>
    <aui-select>
      <aui-option>
         20 
      </aui-option>
      <aui-option>
         50 
      </aui-option>
      <aui-option>
         100 
      </aui-option>
    </aui-select>
  </aui-paginator>
</ng-component>
`;
